networkx/algorithms/isomorphism/VF2pp_helpers/state.py

```diff
@@ -30,3 +30,28 @@ def update_Tinout(G1, G2, T1, T2, T1_out, T2_out, new_node1, new_node2, mapping,
     T2_out = T2_out - uncovered_neighbors_G2
 
     return T1, T2, T1_out, T2_out
+
+
+def restore_Tinout(G1, G2, T1, T2, T1_out, T2_out, popped_node1, popped_node2, mapping, reverse_mapping):
+    # If the node we want to remove from the mapping, has at least one covered neighbor, add it to T1.
+    is_added = False
+    for nbr in G1[popped_node1]:
+        if nbr in mapping:
+            T1.add(popped_node1)
+            is_added = True
+            break
+
+    # Case where the node does not exist in neither the mapping nor T1. By deffinition it should belong to T1_out
+    if not is_added:
+        T1_out.add(popped_node1)
```

Comment:
This looks like a code idiom which is perfect for [the for-else construction](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops).  Python's `for` command can have an `else` clause which is run when the loop finishes normally (without a `break` statement).  Not every user knows about for/else so you want to make sure it is more readable than not using it.  Perhaps something like the following, but you should decide which is more readable:
```python
for nbr in G1[popped_node1]:
    if nbr in mapping:
        T1.add(popped_node1)
        break
else:  # no neighbors in the mapping
    T1_out.add(popped_node1)
```
Here's another form which might or might not be more readable:  (Note that `any` stops the loop when it finds a value that is True just like the `break` syntax does.)
```python
if any(nbr in mapping for nbr in G1[popped_node1]):
    T1.add(popped_node1)
else:
    T1_old.add(popped_node1)
```
And an even shorter, but probably harder to read:
```python
(T1 if any(nbr in mapping for nbr in G1[popped_node1]) else T1_old).add(popped_node1)
```

Reply:
This is actually amazing, didn't have a clue about any of these. Thank you very much. 

Also the restoring of Ti after the removal of a pair from the mapping is also finished and tested (check the unit test and the updated code)

Reply:
Also if you check the updated code, the break statement is gone because ihad to check one more case. So i don't know if i will be able  to use  for-else in the new version..